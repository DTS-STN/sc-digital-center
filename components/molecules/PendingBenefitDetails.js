import React from 'react'
import en from '../../locales/en'
import fr from '../../locales/fr'
import HorizontalRule from '../atoms/HorizontalRule'

const PendingBenefitDetails = (props) => {
  const t = props.locale === 'en' ? en : fr

  return (
    <section>
      <div className="flex-col ml-4 my-4 sm:m-8 sm:flex sm:flex-row"></div>
      <HorizontalRule width="w-auto sm:w-full" />
    </section>
  )
}

export default PendingBenefitDetails
