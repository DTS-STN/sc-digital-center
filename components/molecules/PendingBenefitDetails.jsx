import React from 'react'
import PropTypes from 'prop-types'
import VerticalStepper from '../atoms/VerticalStepper'
import en from '../../locales/en'
import fr from '../../locales/fr'
import HorizontalRule from '../atoms/HorizontalRule'

const PendingBenefitDetails = (props) => {
  const t = props.locale === 'en' ? en : fr

  return (
    <section>
      <div className="flex-col ml-4 my-4 sm:m-8 sm:flex sm:flex-row">
        <VerticalStepper
          benefitStatus={props.benefit.status}
          lastUpdates={props.benefit.lastUpdates}
          locale={props.locale}
        />
        <div id="details" className="mt-4 sm:mt-0 sm:ml-20">
          <div id="application_details" className="mb-5 sm:mb-14">
            <p className="font-bold text-xl">{t.applicationDetails}</p>
            <p className="font-normal text-base">
              {props.benefit.statusDescription}
            </p>
          </div>
          <div id="application_description" className="mb-5 sm:mb-14">
            <p className="font-bold text-base">{t.applicationDescription}</p>
            <p className="text-base">{props.benefit.applicationDescription}</p>
          </div>
          <div id="additionalInformation">
            <p className="font-bold text-base">{t.additionalInformation}</p>
            <p className="text-base">{props.benefit.additionalInformation}</p>
          </div>
        </div>
      </div>
      <HorizontalRule width="w-auto sm:w-full" />
    </section>
  )
}

PendingBenefitDetails.propTypes = {
  benefit: PropTypes.shape({
    benefitType: PropTypes.string,
    benefitName: PropTypes.string,
    status: PropTypes.oneOf(['Active', 'Pending']),
    statusDesciption: PropTypes.string,
    applicationType: PropTypes.string,
    applicationDescription: PropTypes.string,
    additionalInformation: PropTypes.string,
  }),
}

export default PendingBenefitDetails
