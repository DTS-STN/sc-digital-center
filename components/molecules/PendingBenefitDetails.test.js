import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import PendingBenefitDetails from './PendingBenefitDetails'
import { axe } from 'jest-axe'

describe('PendingBenefitDetails Tests', () => {
  const testBenefitObject = {
    benefitType: 'CPP',
    benefitName: 'Canada Pension Plan',
    status: 'Active',
    statusDescription:
      'Your application is pending, we will notify you with decision',
    applicationDate: 'July 1, 2021',
    estimatedReviewCompletionDate: 'July 31, 2021',
    applicationType: 'Canada Pension Plan',
    applicationStatus: 'Application Submitted',
    applicationDescription: 'Online application received',
    nextPaymentAmount: 734.34,
    nextPaymentDate: 'September 30, 2021',
    lastPaymentDate: 'August 30, 2021',
    pensionStartDate: 'August 1, 2021',
    payeeFullName: 'John Smith Doe',
    payeeAddress: '', // 123 - 00 Fake Street, Ottawa, On A1A-1A1
    payeePhoneNumber: '1-613-555-5455',
    accountNumber: 'XXXX-123',
    institutionNumber: '002',
    institutionName: 'Scotiabank',
    payerName: 'Government of Canada',
    paymentDepositDate: 'August 30, 2021',
    paymentStatus: 'Issued',
    paymentType: 'Direct Deposit',
    additionalInformation:
      'We will notify you as soon as we have process your application.',
    pendingBenefits: 'Retirement pension',
  }

  it('renders the details section of a Pending Benefit', () => {
    render(<PendingBenefitDetails benefit={testBenefitObject} />)
  })

  it('has no a11y violations', async () => {
    const { container } = render(
      <PendingBenefitDetails benefit={testBenefitObject} />
    )
    const results = await axe(container)
    expect(results).toHaveNoViolations
  })
})
