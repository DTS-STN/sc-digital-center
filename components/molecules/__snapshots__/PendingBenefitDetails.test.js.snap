// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingBenefitDetails Tests renders the details section of a Pending Benefit 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="flex-col ml-4 my-4 sm:m-8 sm:flex sm:flex-row"
        >
          <div
            class="lg:ml-7"
          >
            <div
              id="stepper-container"
            >
              <h1
                class="text-xl font-bold"
              />
              <div
                class="py-2.5 flex w-60"
              >
                <div
                  class="v-stepper w-5"
                >
                  <div
                    class="w-3 h-3 mt-2 rounded-full ml-1/5 bg-green-active"
                  />
                  <div
                    class="w-0.5 mt-1.5 ml-[5px] h-[92%] bg-green-active"
                  />
                </div>
                <div
                  class="ml-3 "
                >
                  <p
                    class="text-md font-bold"
                  >
                    Documents reviewed
                  </p>
                  <p
                    class="text-sm"
                  >
                    September 8, 2021
                  </p>
                </div>
              </div>
              <div
                class="py-2.5 flex w-60"
              >
                <div
                  class="v-stepper w-5"
                >
                  <div
                    class="w-3 h-3 mt-2 rounded-full ml-1/5 bg-green-active"
                  />
                  <div
                    class="w-0.5 mt-1.5 ml-[5px] h-[92%] bg-green-active"
                  />
                </div>
                <div
                  class="ml-3 "
                >
                  <p
                    class="text-md font-bold"
                  >
                    Documents uploaded
                  </p>
                  <p
                    class="text-sm"
                  >
                    September 6, 2021
                  </p>
                </div>
              </div>
              <div
                class="py-2.5 flex w-60"
              >
                <div
                  class="v-stepper w-5"
                >
                  <div
                    class="w-3 h-3 mt-2 rounded-full ml-1/5 bg-green-active"
                  />
                  <div
                    class="w-0.5 mt-1.5 ml-[5px] h-[92%] bg-green-active"
                  />
                </div>
                <div
                  class="ml-3 "
                >
                  <p
                    class="text-md font-bold"
                  >
                    Documents requested
                  </p>
                  <p
                    class="text-sm"
                  >
                    September 5, 2021
                  </p>
                </div>
              </div>
              <div
                class="pb-2.5 flex mt-0 h-max"
              >
                <div
                  id="v-stepper"
                >
                  <div
                    class="w-0.5 mt-1.5 ml-[5px] h-[92%] bg-green-active"
                  />
                </div>
                <div
                  id="stepper-content"
                >
                  <button
                    class="text-left ml-6 text-xs text-bright-blue-solid underline"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="mt-4 sm:mt-0 sm:ml-20"
            id="details"
          >
            <div
              class="mb-5 sm:mb-14"
              id="application_details"
            >
              <p
                class="font-bold text-xl"
              />
              <p
                class="font-normal text-base"
              >
                Your application is pending, we will notify you with decision
              </p>
            </div>
            <div
              class="mb-5 sm:mb-14"
              id="application_description"
            >
              <p
                class="font-bold text-base"
              />
              <p
                class="text-base"
              >
                Online application received
              </p>
            </div>
            <div
              id="additionalInformation"
            >
              <p
                class="font-bold text-base"
              />
              <p
                class="text-base"
              >
                We will notify you as soon as we have process your application.
              </p>
            </div>
          </div>
        </div>
        <div
          class="w-auto sm:w-full flex-grow border-t border-gray-light mx-4 sm:mx-0"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="flex-col ml-4 my-4 sm:m-8 sm:flex sm:flex-row"
      >
        <div
          class="lg:ml-7"
        >
          <div
            id="stepper-container"
          >
            <h1
              class="text-xl font-bold"
            />
            <div
              class="py-2.5 flex w-60"
            >
              <div
                class="v-stepper w-5"
              >
                <div
                  class="w-3 h-3 mt-2 rounded-full ml-1/5 bg-green-active"
                />
                <div
                  class="w-0.5 mt-1.5 ml-[5px] h-[92%] bg-green-active"
                />
              </div>
              <div
                class="ml-3 "
              >
                <p
                  class="text-md font-bold"
                >
                  Documents reviewed
                </p>
                <p
                  class="text-sm"
                >
                  September 8, 2021
                </p>
              </div>
            </div>
            <div
              class="py-2.5 flex w-60"
            >
              <div
                class="v-stepper w-5"
              >
                <div
                  class="w-3 h-3 mt-2 rounded-full ml-1/5 bg-green-active"
                />
                <div
                  class="w-0.5 mt-1.5 ml-[5px] h-[92%] bg-green-active"
                />
              </div>
              <div
                class="ml-3 "
              >
                <p
                  class="text-md font-bold"
                >
                  Documents uploaded
                </p>
                <p
                  class="text-sm"
                >
                  September 6, 2021
                </p>
              </div>
            </div>
            <div
              class="py-2.5 flex w-60"
            >
              <div
                class="v-stepper w-5"
              >
                <div
                  class="w-3 h-3 mt-2 rounded-full ml-1/5 bg-green-active"
                />
                <div
                  class="w-0.5 mt-1.5 ml-[5px] h-[92%] bg-green-active"
                />
              </div>
              <div
                class="ml-3 "
              >
                <p
                  class="text-md font-bold"
                >
                  Documents requested
                </p>
                <p
                  class="text-sm"
                >
                  September 5, 2021
                </p>
              </div>
            </div>
            <div
              class="pb-2.5 flex mt-0 h-max"
            >
              <div
                id="v-stepper"
              >
                <div
                  class="w-0.5 mt-1.5 ml-[5px] h-[92%] bg-green-active"
                />
              </div>
              <div
                id="stepper-content"
              >
                <button
                  class="text-left ml-6 text-xs text-bright-blue-solid underline"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="mt-4 sm:mt-0 sm:ml-20"
          id="details"
        >
          <div
            class="mb-5 sm:mb-14"
            id="application_details"
          >
            <p
              class="font-bold text-xl"
            />
            <p
              class="font-normal text-base"
            >
              Your application is pending, we will notify you with decision
            </p>
          </div>
          <div
            class="mb-5 sm:mb-14"
            id="application_description"
          >
            <p
              class="font-bold text-base"
            />
            <p
              class="text-base"
            >
              Online application received
            </p>
          </div>
          <div
            id="additionalInformation"
          >
            <p
              class="font-bold text-base"
            />
            <p
              class="text-base"
            >
              We will notify you as soon as we have process your application.
            </p>
          </div>
        </div>
      </div>
      <div
        class="w-auto sm:w-full flex-grow border-t border-gray-light mx-4 sm:mx-0"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
